lib/pages/interpretation_page.dart

```diff
@@ -0,0 +1,237 @@
+class InterpretationWidget extends StatefulWidget {
+  final BluetoothDevice device;
+
+  const InterpretationWidget({Key? key, required this.device})
+      : super(key: key);
+
+  @override
+  _InterpretationWidgetState createState() =>
+      _InterpretationWidgetState(device);
+}
+
+class _InterpretationWidgetState extends State<InterpretationWidget> {
+  static final String TAG = "InterpretationWidget";
+  late Stream<List<int>> interpretationStream;
+  static final Color background = Color.fromRGBO(0xD3, 0xD3, 0xD3, 1.0);
+  final BluetoothDevice device;
+
+  _InterpretationWidgetState(this.device);
+
+  @override
+  void initState() {
+    super.initState();
+    loadInterpretationStream();
+  }
+
+  Future<void> loadInterpretationStream() async {
+    interpretationStream = new Stream.empty();
+    interpretationStream = await BluetoothBackend.getLsaGlovesService(device)
+        .then((service) =>
+            BluetoothBackend.getInterpretationCharacteristic(service!))
+        .then((characteristic) {
+      characteristic.setNotifyValue(true);
+      return characteristic.value;
+    });
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    developer.log("Disposed widget of device: " + device.id.id, name: TAG);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        width: double.infinity,
+        child: Column(
+          children: <Widget>[
+            Container(
+                width: double.infinity,
+                padding: const EdgeInsets.all(8.0),
+                alignment: Alignment.topCenter,
+                color: background,
+                child: Text(device.name)),
+            Container(
+                width: double.infinity,
+                padding: const EdgeInsets.all(8.0),
+                alignment: Alignment.topCenter,
+                color: background,
+                child: Text("Mac addr: ${device.id.id}")),
+            displayStats()
+          ],
+        ));
+  }
+
+  StreamBuilder<List<int>> displayStats() {
+    return StreamBuilder<List<int>>(
+        stream: interpretationStream,
+        builder: (c, rawDeviceInterpretations) {
+          String msg = "";
+          if (rawDeviceInterpretations.hasData) {
+            msg = new String.fromCharCodes(rawDeviceInterpretations.data!);
+          }
+          return Container(
+              width: double.infinity,
+              height: 80,
+              alignment: Alignment.center,
+              color: background,
+              child: Text(msg));
+        });
+  }
+}
+
+class InterpretationButton extends StatefulWidget {
+  const InterpretationButton({Key? key}) : super(key: key);
+
+  @override
+  _InterpretationButtonState createState() => _InterpretationButtonState();
+}
+
+class _InterpretationButtonState extends State<InterpretationButton> {
+  bool _isEnabled = false;
+  bool _isRunning = false;
+  late StreamSubscription<List<BluetoothDevice>> streamSubscription;
```

Comment:
Porque en lugar de hacerlo late no lo inicializas en el constructor?

Reply:
No me gusta la idea de inicializar elementos asociados al estado del widget en el constructor, me parece que el lugar indicado es el initState, similar al onCreate de Android nativo (donde nunca inicializás cosas en el constructor de una activity por ejemplo), y esto es porque cuando se ejecuta el initState (habiéndose creado ya el objeto) tenemos acceso al BuildContext y al StatefulWidget asociado a este estado (según este artículo https://flutteragency.com/difference-between-initstate-and-a-class-constructor-in-flutter/) y por más que acá no necesitemos de esos elementos para inicializar el StreamSubscription, conceptualmente me parece correcto que esté en el initState y no en el constructor, y el `late` está para permitirnos eso.